tests/test_receivers.py

```diff
@@ -165,3 +165,35 @@ def test_event_deletion(app, receiver):
         event.delete()
         assert event.response == {'status': 410, 'message': 'Gone.'}
         assert event.response_code == 410
+
+
+def test_event_status(app):
+    """Test event status reporting."""
+    data = json.dumps({'x': 40, 'y': 2})
+
+    class TestCeleryReceiver(CeleryReceiver):
+
+        def run(self, event):
+            """Change task state."""
+            from celery import current_task, states
+
+            event.response['message'] = event.payload['x'] + event.payload['y']
```

Comment:
If I move the addition operation to a separate Celery task, I get this error:
```python
[2016-10-14 17:17:12,407] ERROR in models: Could not process event.
Traceback (most recent call last):
  File "/home/orestis/.virtualenvs/invenio-webhooks/src/invenio-webhooks/invenio_webhooks/models.py", line 268, in process
    self.receiver(self)
  File "/home/orestis/.virtualenvs/invenio-webhooks/src/invenio-webhooks/invenio_webhooks/models.py", line 171, in __call__
    process_event.apply_async(task_id=str(event.id), args=[event.id])
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/task.py", line 557, in apply_async
    link=link, link_error=link_error, **options)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/task.py", line 755, in apply
    request=request, propagate=throw)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/trace.py", line 355, in eager_trace_task
    uuid, args, kwargs, request)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/trace.py", line 253, in trace_task
    I, R, state, retval = on_error(task_request, exc, uuid)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/flask_celeryext/app.py", line 43, in __call__
    return Task.__call__(self, *args, **kwargs)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/celery/app/task.py", line 420, in __call__
    return self.run(*args, **kwargs)
  File "/home/orestis/.virtualenvs/invenio-webhooks/src/invenio-webhooks/invenio_webhooks/models.py", line 149, in process_event
    db.session.add(event)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 482, in __exit__
    self._assert_active(deactive_ok=True, prepared_ok=True)
  File "/home/orestis/.virtualenvs/invenio-webhooks/local/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 223, in _assert_active
    raise sa_exc.ResourceClosedError(closed_msg)
ResourceClosedError: This transaction is closed
```

Therefore, I cannot run other tasks from the master task (i.e. the `CeleryReceiver::run` method).
Do you know why this happens?